core/src/main/java/com/datastrato/gravitino/catalog/BaseCatalog.java

```diff
@@ -167,4 +173,12 @@ public Audit auditInfo() {
     Preconditions.checkArgument(entity != null, "entity is not set");
     return entity.auditInfo();
   }
+
+  protected CatalogOperations asProxyOps(CatalogOperations ops, CatalogProxyPlugin plugin) {
+    return CatalogOperationsProxy.getProxy(ops, plugin);
+  }
+
+  protected CatalogProxyPlugin newProxyPlugin(Map<String, String> config) {
```

Comment:
Can we make `CatalogProxyPlugin` invisible to detailed catalogs? Making it as a method of catalog seems too invasive?

Reply:
Every catalog should be proxied actually. So I choose to move CatalogProxyPlugin to here.